common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -233,4 +233,49 @@ public static String getErrorMessage(String errorMessage, String modelId, Boolea
             return errorMessage + " Model ID: " + modelId;
         }
     }
+
+    /**
+     * Collects the prefixes of the toString() method calls present in the values of the given map.
+     *
+     * @param map A map containing key-value pairs where the values may contain toString() method calls.
+     * @return A list of prefixes for the toString() method calls found in the map values.
+     */
+    public static List<String> collectToStringPrefixes(Map<String, String> map) {
+        List<String> prefixes = new ArrayList<>();
+        for (String key : map.keySet()) {
+            String value = map.get(key);
+            if (value != null) {
+                Pattern pattern = Pattern.compile("\\$\\{(\\w+\\.\\w+)\\.toString\\(\\)\\}");
+                Matcher matcher = pattern.matcher(value);
+                while (matcher.find()) {
+                    String prefix = matcher.group(1);
+                    prefixes.add(prefix.substring(prefix.lastIndexOf('.') + 1));
+                }
+            }
+        }
+        return prefixes;
+    }
+
+    /**
+     * Parses the given parameters map and processes the values containing toString() method calls.
+     *
+     * @param parameters A map containing key-value pairs where the values may contain toString() method calls.
+     * @return A new map with the processed values for the toString() method calls.
+     */
+    public static Map<String, String> parseParameters(Map<String, String> parameters) {
```

Comment:
Add prefix to method parameter list ?  For connector, it will be `parameters`. For other usecase, it maybe different, like local model could be `model_config`
```suggestion
    public static Map<String, String> parseParameters(String prefix, Map<String, String> parameters) {
```

Reply:
all parameters will be prefix with `parameters` in   here 

https://github.com/opensearch-project/ml-commons/blob/59b417be215290c9af19bb2e4e1a4bfe5d007ec7/common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java#L327

I added the test to test when it's `parameters.model_config.context`, then it would be taking the prefix as `model_config.context` properly, please see this UT 

https://github.com/opensearch-project/ml-commons/blob/59b417be215290c9af19bb2e4e1a4bfe5d007ec7/common/src/test/java/org/opensearch/ml/common/utils/StringUtilsTest.java#L316-L340